src/main/rx/rx.c

```diff
@@ -245,6 +245,29 @@ uint16_t calculateNonDataDrivenChannel(uint8_t chan, uint16_t sample)
     return rcDataMean[chan] / PPM_AND_PWM_SAMPLE_COUNT;
 }
 
+uint16_t applyRxCalibration(int sample, int minrc, int maxrc)
+{
+    int extended_midrc = minrc + (maxrc - minrc) / 2;
+    int extended_minrc = (minrc - extended_midrc) * (PULSE_MAX - PULSE_MIN) / (PWM_RANGE_MAX - PWM_RANGE_MIN) + extended_midrc;
+    int extended_maxrc = (maxrc - extended_midrc) * (PULSE_MAX - PULSE_MIN) / (PWM_RANGE_MAX - PWM_RANGE_MIN) + extended_midrc;
+    
+    // Check special condition sample = PPM_RCVR_TIMEOUT, allow no signal value to avoid being corrupted by calibration
+    if (sample == 0) {
+        return 0;
+    }
+
+    // Check extreme conditions
```

Comment:
using scaleRange() [https://github.com/cleanflight/cleanflight/blob/master/src/main/common/maths.c#L89] and then saturating to min/max may be easier to read